### ISSUE TITLE
The posix implementation of os.listdir() filters all files starting with two periods

### ISSUE THREAD

**Issue by jeremysrand** [micropython/micropython#11032]
I am using the UNIX port of micro python, v1.19.  If I do the following in the shell:

`$ touch .testing ..testing ...testing`

And then run this from the REPL or a script in the same directory:

```
import os
for file in os.listdir():
    print(file)
```

The output will show .testing but not the files ..testing or ...testing.

The problem is caused by this line in extmod/vfs_posix.c:

```
        if (fn[0] == '.' && (fn[1] == 0 || fn[1] == '.')) {
            // skip . and ..
            continue;
        }
```

This code doesn't check that fn[2] is a NUL char.  So, as long as the first two chars are a period, the file will be skipped.

**Comment by dlech** [comment]
Sounds like an easy fix. Would you like to make a pull request?

**Comment by jeremysrand** [comment]
Sure.  It does look like a trivial fix.  I will work on a pull request.

**Comment by dpgeorge** [comment]
Fixed by d677023b3decec2f10d6b00b84b34032db4a0fd4
